src/listener/tcp_listener.rs

```diff
@@ -4,44 +4,96 @@ use crate::listener::Listener;
 use crate::{log, Server};
 
 use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+
+use std::net::ToSocketAddrs;
 
 use async_std::net::{self, SocketAddr, TcpStream};
 use async_std::prelude::*;
 use async_std::{io, task};
+use http_types::Url;
 
 /// This represents a tide [Listener](crate::listener::Listener) that
```

Comment:
this block of documentation is many commits out of date and needs to be updated before merge